plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ptf/Procedure.java

```diff
@@ -0,0 +1,136 @@
+{
+    public static final String SCHEMA_NAME = "system";
+    public static final String NAME = "procedure";
+
+    private final JdbcTransactionManager transactionManager;
+
+    @Inject
+    public Procedure(JdbcTransactionManager transactionManager)
+    {
+        this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
+    }
+
+    @Override
+    public ConnectorTableFunction get()
+    {
+        return new ClassLoaderSafeConnectorTableFunction(new ProcedureFunction(transactionManager), getClass().getClassLoader());
+    }
+
+    public static class ProcedureFunction
+            extends AbstractConnectorTableFunction
+    {
+        private final JdbcTransactionManager transactionManager;
+
+        public ProcedureFunction(JdbcTransactionManager transactionManager)
+        {
+            super(
+                    SCHEMA_NAME,
+                    NAME,
+                    List.of(ScalarArgumentSpecification.builder()
+                            .name("QUERY")
+                            .type(VARCHAR)
+                            .build()),
+                    GENERIC_TABLE);
+            this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
+        }
+
+        @Override
+        public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+        {
+            ScalarArgument argument = (ScalarArgument) getOnlyElement(arguments.values());
+            String query = ((Slice) argument.getValue()).toStringUtf8();
+
+            JdbcMetadata metadata = transactionManager.getMetadata(transaction);
+            JdbcProcedureHandle tableHandle = metadata.getProcedureHandle(session, new ProcedureQuery(query));
+            List<JdbcColumnHandle> columns = tableHandle.getColumns();
+            Descriptor returnedType = new Descriptor(columns.stream()
+                    .map(column -> new Field(column.getColumnName(), Optional.of(column.getColumnType())))
+                    .collect(toImmutableList()));
+
+            ProcedureFunctionHandle handle = new ProcedureFunctionHandle(tableHandle);
+
+            return TableFunctionAnalysis.builder()
+                    .returnedType(returnedType)
+                    .handle(handle)
+                    .build();
+        }
+    }
+
+    public static class ProcedureFunctionHandle
+            implements ConnectorTableFunctionHandle
+    {
+        private final JdbcProcedureHandle tableHandle;
+
+        @JsonCreator
+        public ProcedureFunctionHandle(@JsonProperty("tableHandle") JdbcProcedureHandle tableHandle)
+        {
+            this.tableHandle = requireNonNull(tableHandle, "tableHandle is null");
+        }
+
+        @JsonProperty
+        public ConnectorTableHandle getTableHandle()
+        {
+            return tableHandle;
+        }
+    }
+
+    public record ProcedureQuery(@JsonProperty String query)
```

Comment:
not sure if we use records for between-nodes serialization (I think we are starting to)

then `ProcedureFunctionHandle` above can be a record too

also, i don't think we need `@JsonProperty` and `@JsonCreator` for records (https://github.com/airlift/airlift/pull/993), do we?

Reply:
`ProcedureFunctionHandle` - wanted be in-sync with other `TableHandle` implementation. I don't have any strong opinions on it. Will move it to a record.